compiler/src/dotty/tools/dotc/cc/CaptureOps.scala

```diff
@@ -445,6 +445,16 @@ extension (tp: AnnotatedType)
     case ann: CaptureAnnotation => ann.boxed
     case _ => false
 
+class CleanupRetains(using Context) extends TypeMap:
+  def apply(tp: Type): Type = cleanupRetains(tp, this)
+
+/** Drop retains annotations in the type. */
+def cleanupRetains(tp: Type, theMap: CleanupRetains | Null = null)(using Context): Type =
```

Comment:
You don't need the `theMap` trick. Just define a normal typemap like:

```scala
class CleanupRetains(using Context) extends TypeMap:
   def apply(tp: Type): Type = tp match
     case RetainingType(tp, _) => tp
     case _ => mapOver(tp)
```
and use that instead of the `cleanupRetains` method in PostTyper.

The `theMap` trick is used if we want to _avoid_ creating the type map for simple types because the operation is very hot. But to pay off, you then need to actually handle simple types in the associated methods. Here, cleanupRetains is fairly warm but not as hot as methods that use the theMap trick such as substitutions.